R/sass.R

```diff
@@ -142,6 +149,47 @@ sass <- function(input = NULL, options = sass_options(), output = NULL,
   css
 }
 
+#' An intelligent (temporary) output file
+#'
+#' Intended for use with [sass()]'s `output` argument for temporary file
+#' generation that is `cache` and `options` aware. In particular, this ensures
+#' that new redundant file(s) aren't generated on a [sass()] cache hit, and that
+#' the file's extension is suitable for the [sass_options()]'s `output_style`.
+#'
+#' @param basename a non-empty character vector giving the outfile name (without
+#'   the extension).
+#' @param dirname a non-empty character vector giving the initial part of the
+#'   directory name.
+#' @param fileext the output file extension. The default is `".min.css"` for
+#'   compressed and compact output styles; otherwise, its `".css"`.
+#'
+#' @return A function with two arguments: `options` and `suffix`. When called inside
+#' [sass()] with caching enabled, the caching key is supplied to `suffix`.
+#'
+#' @export
+#' @examples
+#' sass("body {color: red}", output = output_file())
+#'
+#' func <- output_file(basename = "foo", dirname = "bar-")
+#' func(suffix = "baz")
+#'
+output_file <- function(basename = "sass", dirname = basename, fileext = NULL) {
```

Comment:
`css_output_file`?

Reply:
Renamed in https://github.com/rstudio/sass/pull/48/commits/cb5082cb38a6d50078a4d1104e700d13ee4eaac9